### DISCUSSION TITLE
Pydantic v2 validation error for query parameter model is not turned into 422 error

### DISCUSSION THREAD

**Discussion by chbndrhnns** [fastapi/fastapi#10336] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

import fastapi
import pydantic
from starlette.testclient import TestClient

app = fastapi.FastAPI()


class Pagination(pydantic.BaseModel):
    skip: int = 0
    limit: Annotated[int, pydantic.Field(ge=1)] = 1


@app.get("/")
async def get(pagination: Pagination = fastapi.Depends()):
    return {"data": pagination}


def test_():
    with TestClient(app) as client:
        res = client.get("/?limit=-1")
        assert res.status_code == 422
```

### Description

* Run test

It passes for pydantic v1 and it fails for pydantic v2

I expect FastAPI to consistently turn validation errors into status code 422.
That does not happen if a pydantic v2 model is used directly as a dependency in a path.

### Operating System

macOS

### FastAPI Version

0.103.1

### Pydantic Version

2.4.0

### Python Version

3.9.15

**Comment by ebottos94** [comment]
Hi @chbndrhnns, this is not the correct way to implement query parameters validation : https://fastapi.tiangolo.com/tutorial/query-params-str-validations/

```python
from typing import Annotated
from fastapi import FastAPI, Query
from starlette.testclient import TestClient

app = FastAPI()

@app.get("/")
async def endpoint(
   skip: int = 0, 
   limit : Annotated[int, Query(ge=1)] = 1
):
    response = {
       "skip" : skip, 
       "limit" : limit
    }
    return response

def test_endpoint():
    with TestClient(app) as client:
        res = client.get("/?limit=-1")
        assert res.status_code == 422
```

**Reply by chbndrhnns** [reply]
I'd like a single model to capture my query parameters and there are couple of examples out there that show that this is possible([official docs](https://fastapi.tiangolo.com/tutorial/dependencies/classes-as-dependencies/#shortcut), [stackoverflow](https://stackoverflow.com/a/62468403/6112272)). 
Furthermore, the snippet works when using a pydantic v1 model (use `pydantic.v1.BaseModel`) which is why I consider this a regression.

**Reply by ebottos94** [reply, marked as answer]
This seems to work : 

```python
from fastapi import FastAPI, Depends, Query
from fastapi.testclient import TestClient
from pydantic import BaseModel, Field

app = FastAPI()


class Pagination(BaseModel):
    skip: int = 0
    limit: int | None = Field(Query(default=1, ge=1))


@app.get("/")
async def endpoint(pagination: Pagination = Depends()):
    return {"data": pagination}

def test_():
    with TestClient(app) as client:
        res = client.get("/?limit=-1")
        assert res.status_code == 422
```

**Reply by chbndrhnns** [reply]
Thanks, this is working nicely for me. I was not aware that I can wrap `fastapi.Query` with `pydantic.Field`.
